### PR TITLE
fix(scrypt): allow empty or missing salt for backward compatibility and improve hex parsing validation

### PR DESCRIPTION
**Pull request by sergei-boiko-trustwallet**
This pull request improves the handling and validation of hexadecimal and scrypt parameters, particularly for backward compatibility with keystore files that have empty or missing salts. It introduces a safer hex parsing function, updates scrypt parameter validation to allow for older key formats, and adds comprehensive tests to ensure correct behavior.

**Hexadecimal parsing improvements:**
- Added a new `parse_hex_checked` function in `HexCoding.h` that safely parses hex strings and returns a `Result<Data>` indicating success or failure, improving error handling and input validation.
- Refactored the hex string padding logic into a dedicated `pad_left_hex` function for clarity and reuse.
- Updated tests in `HexCodingTests.cpp` to cover new hex parsing logic, including edge cases for empty, odd-length, and invalid hex strings.

**Scrypt parameter compatibility and validation:**
- Modified `ScryptParameters` to allow empty or missing salt fields for backward compatibility with existing keystore files, defaulting to an empty salt if not present.
- Updated validation to only reject salts that exceed the maximum allowed length, not those that are empty or too short.

**Keystore loading and test coverage:**
- Added test vectors and tests to ensure keystore files with empty or missing salt fields are loaded and decrypted correctly, and that the JSON output always includes a (possibly empty) salt field.

### CODE REVIEW

**Review by BSCSecChef — DISMISSED** [review]
LGTM

File: src/HexCoding.h [hunk 1]
```diff
@@ -76,20 +77,34 @@ inline std::string hex(uint64_t value) {
...
+/// Pads a hexadecimal string with a leading zero if it has an odd length and the `padLeft` flag is set.
+inline std::string pad_left_hex(const std::string& string, bool padLeft = false) {
     if (string.size() % 2 != 0 && padLeft) {
         std::string temp = string;
         if (temp.compare(0, 2, "0x") == 0) {
             temp.erase(0, 2);
         }
         temp.insert(0, 1, '0');
-        return internal::parse_hex(temp);
+        return temp;
```
[the remark this answers is not in the document]
Comment (sergei-boiko-trustwallet) [reply src/HexCoding.h]: I made a fix in `pad_left_hex` function, so it no longer erases `0x` prefix if presents

File: tests/common/HexCodingTests.cpp [hunk 2]
```diff
@@ -33,4 +33,59 @@ TEST(HexCoding, isHexEncoded) {
...
+
+    // Empty string
+    res = parse_hex_checked("");
+    ASSERT_TRUE(res.isSuccess());
+    EXPECT_TRUE(res.payload().empty());
+
+    // 0x string
+    res = parse_hex_checked("0x");
+    ASSERT_TRUE(res.isSuccess());
+    EXPECT_TRUE(res.payload().empty());
```
[the remark this answers is not in the document]
Comment (sergei-boiko-trustwallet) [reply tests/common/HexCodingTests.cpp:55]: `0x` is a valid hex string, so the issue is in `is_hex_encoded` function. However, this is out of the PR scope.

**Review by nikhil-gupta-tw — APPROVED** [review]

**Review by BSCSecChef — APPROVED** [review]
LGTM
